vllm/engine/async_llm_engine.py

```diff
@@ -479,6 +491,37 @@ async def check_health_async(self) -> None:
             self.tokenizer.check_health()
         self.model_executor.check_health()
 
+    async def _build_guided_decoding_logits_processor_async(
+            self, sampling_params: SamplingParams,
+            lora_request: Optional[LoRARequest]) -> SamplingParams:
+        """Constructs logits processors based on the guided_decoding,
+        logits_bias, and allowed_token_ids fields in sampling_params. Deletes
+        those fields and adds the constructed logits processors to the
+        logits_processors field. Returns the modified sampling params."""
+        if (guided_decoding := sampling_params.guided_decoding) is None:
+            return sampling_params
+
+        logger.debug(
+            "Building guided decoding logits processor in "
+            "AsyncLLMEngine. Params: %s", guided_decoding)
+
+        tokenizer = self.get_tokenizer(lora_request=lora_request)
```

Comment:
```suggestion
        tokenizer = await self.get_tokenizer(lora_request=lora_request)
```

Reply:
oof- looks like that's missing a test somewhere then? :(

Reply:
Ah, actually this calls the non-async method from llm_engine, so this was covered by the tests and was originally correct